backend/src/v1/services/auth-service.ts

```diff
@@ -243,16 +261,22 @@ const auth = {
       log.error(`no bceid_user_guid found in the jwt token`, userInfo.jwt);
       res.redirect(config.get('server:frontend') + '/login-error');
     }
-    if(!req.session?.companyDetails){
-      try{
-        req.session.companyDetails = await getCompanyDetails(userGuid);
-        await auth.storeUserInfo(req, userInfo);
-      }catch (e) {
-        log.error(`Error happened while getting company details from BCEID for user ${userGuid}`, e);
-        res.redirect(config.get('server:frontend') + '/login-error');
+    try {
+      auth.validateClaims(userInfo.jwt);
+      if(!req.session?.companyDetails){
+        try{
+          req.session.companyDetails = await getCompanyDetails(userGuid);
+          await auth.storeUserInfo(req, userInfo);
+        }catch (e) {
+          log.error(`Error happened while getting company details from BCEID for user ${userGuid}`, e);
+          res.redirect(config.get('server:frontend') + '/login-error');
+        }
       }
+      res.redirect(config.get('server:frontend'));
+    }catch (e) {
+      log.error('invalid claims in token', e);
+      res.redirect(config.get('server:frontend') + '/login-error');
```

Comment:
This catch block seems specific to auth.validateClaims(...).  Why include the IF block within tyr try?